### DISCUSSION TITLE
Example for how to do a LoadingView for any long-running operation

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno.toolkit.ui#579] [Q&A]
Is there an example for how I can use a LoadingView to display an overlay of a progress ring, perhaps added to the Shell, with a method that I can call to tell it to display/hide the progress ring? So I can have a global LoadingView that can be invoked at multiple places/pages.

**Comment by kazo0** [comment]
Hi @HakanL!

I don't believe there is a specific example for this scenario but, if it helps, here are the current overview docs for the LoadingView: https://platform.uno/docs/articles/external/uno.toolkit.ui/doc/controls/LoadingView.html

For your case, I would imagine that you could add a LoadingView into your Shell.xaml and bind it to a property in your view model for your Shell that returns an ILoadable. You could then utilize the [Uno.Extensions.Reactive.Messaging](https://www.nuget.org/packages/Uno.Extensions.Reactive.Messaging/2.4.0-dev.323) package to be able to send a message from any of your view models for the Shell to execute the ILoadable. The Messaging package is using the [WinUI Community Toolkit Messenger](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/messenger) so you would be able to follow their APIs 

You may need to overlay this LoadingView on top of the root Frame that your Shell uses for navigation, perhaps @nickrandolph may be able to explain that part better than me?

**Reply by HakanL** [reply]
Thanks for pointing me in the right direction. The internal communication to display/hide the view I'm not worried about, but I'm coming up short on how to add the view and make it appear on top of everything.

**Reply by nickrandolph** [reply]
I would typically avoid putting loadingview at root (with the exception of extended splash screen that is a loadingview). Putting at root conceptually sounds a good idea but I'm yet to see a great app that makes this work. Better the keep loading view for smaller segments of the UI.
Also, please try to use the FeedView with MVUX where possible. This already has a Loadingview!!

**Reply by HakanL** [reply]
I've looked into MVUX but opted to not use it as it didn't seem to fit my needs as well as a regular MVVM. But could you point me to the files where the LoadingView is implemented then maybe I can use it as a template? My UX will mainly consist of menus with a few child options, nothing crazy.
